networkit/graph.pyx

```diff
@@ -665,6 +665,220 @@ cdef class Graph:
 			yield dereference(it)
 			preincrement(it)
 
+	def attachNodeAttribute(self, name, ofType):
+		"""
+		attachNodeAttribute(name, ofType)
+
+		Attaches a node attribute to the graph and returns it.
+
+		.. code-block::
+			
+			A = G.attachNodeAttribute("attributeIdentifier", ofType)
+		
+		All values are initially undefined for existing nodes values can be set/get
+		by 
+		
+		.. code-block:: 
+		
+			A[node] = value # set
+			value = A[node] # get
+
+		Getting undefined values raises a ValueError removing a node makes all
+		its attributes undefined
+
+		Parameters
+		----------
+		name   : str
+			Name for this attribute
+		ofType : type
+			Type of the attribute (either int, float, or str)
+
+		Returns
+		-------
+		networkit.graph.NodeAttribute
+			The resulting node attribute container.
+		"""
+		if not isinstance(name, str):
+			raise Exception("Attribute name has to be a string")
+
+		if ofType == int:
+			return NodeAttribute(NodeIntAttribute().setThis(self._this.attachNodeIntAttribute(name.encode('UTF-8')), &self._this), int)
+		elif ofType == float:
+			return NodeAttribute(NodeDoubleAttribute().setThis(self._this.attachNodeDoubleAttribute(name.encode('UTF-8')), &self._this), float)
+		elif ofType == str:
+			return NodeAttribute(NodeStringAttribute().setThis(self._this.attachNodeStringAttribute(name.encode('UTF-8')), &self._this), str)
```

Comment:
Instead of `.encode('UTF-8')` we can use the `stdstring` function from `helpers`, I will push a fix for this.

Reply:
Thanks! This is something, which is done in several modules actually (re-implementing instead of using `helpers`). Already on my agenda to fix this in an upcoming PR :)